codegenerator/cli/templates/static/codegen/src/GqlDbCustomTypes.res

```diff
@@ -7,12 +7,29 @@ module Float = {
     S.string
     ->S.setName("GqlDbCustomTypes.Float")
     ->S.transform(s => {
-      parser: (. string) => {
+      parser: string => {
         switch string->Belt.Float.fromString {
         | Some(db) => db
-        | None => s.fail(. "The string is not valid GqlDbCustomTypes.Float")
+        | None => s.fail("The string is not valid GqlDbCustomTypes.Float")
         }
       },
-      serializer: (. float) => float->Js.Float.toString,
+      serializer: float => float->Js.Float.toString,
     })
 }
+
+// Schema allows parsing strings or numbers to ints
+// this is needed for entity_history field where we on the first copy we encode all numbers as strings
+// to avoid loss of precision. Otherwise we always serialize to int
+module Int = {
+  @genType
+  type t = int
+
+  external number: unknown => Js.Json.t = "Number"
+
+  let schema = S.custom("GqlDbCustomTypes.Int", _ => {
+    parser: unknown => {
+      unknown->number->S.parseOrRaiseWith(S.int)
+    },
+    serializer: int => int,
+  })
+}
```

Comment:
This is going to be more performant:
```
let schema = S.union([
  S.int,
  S.string->S.transform(s => {parser: string => string->Js.Int.unsafeFromString})
])->S.setName("GqlDbCustomTypes.Int")
```

Reply:
`Js.Int.unsafeFromString` doesn't exist... Is this a util? Or should I just use "Number()" for this?

Reply:
It's pseudocode. `Number()` should work here